### ISSUE TITLE
Serialize Alexa Request

### ISSUE THREAD

**Issue by bauchschuss** [stoiveyp/Alexa.NET.Gadgets#12]
Hi,
I've a problem serializing the Alexa Request. I've been writing the request to CloudWatch this since using Alexa.NET library. In my newest project i'm working with Echo buttons and therefore the Alexa.NET Gadgets library. 

The first request can be serialized and logged without any problems. In the first response i'm doing a roll call. When pressing my two echo buttons there are recongized and sent back to my skill with the second request which i'm not able to serialize using JSonSerializer.

Maybe someone can help me because during development it is really helpful to write the complete request string into a cloud watch stream.

```
private void LogRequestResponse(object requestResponse, string title)
        {
            var serializer = new Amazon.Lambda.Serialization.Json.JsonSerializer();

            using (var ms = new MemoryStream())
            {
                try
                {
                    serializer.Serialize(requestResponse, ms);
                    var jsonText = Encoding.UTF8.GetString(ms.ToArray());

                    Log.LogLine(string.Format("{0} - {1}", title, jsonText));
                }
                catch (Exception e)
                {
                    ErrorFormat("Error serializing {0}, Error: {1}", title, e.ToString());
                }
            }
        }
```

this leads to the following Exception:

> ERROR - Error serializing REQUEST, Error: Newtonsoft.Json.JsonWriterException: Token PropertyName in state Property would result in an invalid JSON object. Path 'request.events[0].inputEvents[0]'.
at Newtonsoft.Json.JsonWriter.AutoComplete(JsonToken tokenBeingWritten)
at Newtonsoft.Json.JsonTextWriter.WritePropertyName(String name, Boolean escape)
at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.SerializeObject(JsonWriter writer, Object value, JsonObjectContract contract, JsonProperty member, JsonContainerContract collectionContract, JsonProperty containerProperty)
at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.SerializeList(JsonWriter writer, IEnumerable values, JsonArrayContract contract, JsonProperty member, JsonContainerContract collectionContract, JsonProperty containerProperty)
at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.SerializeObject(JsonWriter writer, Object value, JsonObjectContract contract, JsonProperty member, JsonContainerContract collectionContract, JsonProperty containerProperty)
at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.SerializeList(JsonWriter writer, IEnumerable values, JsonArrayContract contract, JsonProperty member, JsonContainerContract collectionContract, JsonProperty containerProperty)
at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.SerializeObject(JsonWriter writer, Object value, JsonObjectContract contract, JsonProperty member, JsonContainerContract collectionContract, JsonProperty containerProperty)
at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.SerializeObject(JsonWriter writer, Object value, JsonObjectContract contract, JsonProperty member, JsonContainerContract collectionContract, JsonProperty containerProperty)
at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.Serialize(JsonWriter jsonWriter, Object value, Type objectType)
at Newtonsoft.Json.JsonSerializer.SerializeInternal(JsonWriter jsonWriter, Object value, Type objectType)
at Amazon.Lambda.Serialization.Json.JsonSerializer.Serialize[T](T response, Stream responseStream)

**Comment by stoiveyp** [comment]
Hi @bauchschuss 

Thanks for using the gadget library. The reason is that the InputEvents object, for consistency, use the Alexa.NET MixedDateConverter. This doesn't output anything when being serialized, which means if you try to re-serialize anything using it, you'll get an error.

Actually, the RequestConverter specifically states that requests can't be written - https://github.com/timheuer/alexa-skills-dotnet/blob/81d6418c9baee77da82be8b52b6a942015685c7a/Alexa.NET/Request/Type/RequestConverter.cs#L20

I'll raise a PR in Alexa.NET to update the converter, and if it's approved I'll update the Alexa.NET reference in the gadget library to match.

**Comment by stoiveyp** [comment]
Tim Heuer pushed the fix, and I've updated the reference to 1.5.1 so hopefully it should be fixed now.
Please confirm when you get the chance.

**Comment by stoiveyp** [comment]
This is fixed with the latest Alexa.NET so I'm closing the issue
